common/src/sync.php

```diff
@@ -0,0 +1,110 @@
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class sync extends ModuleFrontController {
+
+    /** @var bool */
+    public $ajax;
+
+    /**
+     * @throws Exception
+     */
+    public function postProcess() {
+        $request_data = trim(file_get_contents('php://input'));
+        if (!$request_data) {
+            throw new HttpException(400, 'Empty request data');
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            throw new HttpException(400, 'Invalid JSON data provided');
+        }
+
+        if (
+            !$this->hasCredentialFields($request_data['webshop_id'], $request_data['api_key'])
+            || $this->credentialsEmpty($request_data['webshop_id'], $request_data['api_key'])
+        ) {
+            throw new HttpException(403, 'Missing credential fields');
+        }
+        $lang_id = (int) Configuration::get('PS_LANG_DEFAULT');
+        $product = new Product($request_data['product_review']['product_id'], false, $lang_id);
+        if (!Validate::isLoadedObject($product)) {
+            throw new HttpException(404, sprintf('Could not find product with ID (%d)', $request_data['product_review']['product_id']));
+        }
+
+        if (!Configuration::get(strtoupper($this->module->name) . '_SYNC_PROD_REVIEWS')) {
+            throw new HttpException(403, 'Product review sync is disabled.');
+        }
+
+        $this->isAuthorized($request_data['webshop_id'], $request_data['api_key']);
+        $this->syncProductReview($request_data['product_review']);
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function syncProductReview(array $product_review) {
+        $this->ajax = 1;
+        if (
+            !$product_review['title']
+            || !$product_review['review']
+            || !$product_review['reviewer']['name']
+            || !$product_review['rating']
+        ) {
+            throw new HttpException(404, 'Missing required content for product review');
+        }
+        $entity_manager = $this->container->get('doctrine.orm.entity_manager');
+        $product_comment_entity = new ProductComment();
+        $date_add = DateTime::createFromFormat('Y-m-d H:i:s', date('Y-m-d H:i:s'));
+        $product_comment_entity->setProductId($product_review['product_id'])
+            ->setCustomerId(0)
+            ->setGuestId(0)
+            ->setTitle($product_review['title'])
+            ->setContent($product_review['review'])
+            ->setCustomerName($product_review['reviewer']['name'])
+            ->setGrade($product_review['rating'])
+            ->setValidate(1)
+            ->setDeleted($product_review['deleted'] ? 1 : 0)
+            ->setDateAdd($date_add);
+        $entity_manager->persist($product_comment_entity);
+        $entity_manager->flush();
+
+        $review_id = $product_comment_entity->getId();
+        if (!$review_id) {
+            $this->ajaxRender('There has been an error syncing the review in PrestaShop');
```

Comment:
Here you should return an error, instead of just some text. 
It is a communication between 2 API endpoints, not 2 people.